### PR TITLE
Change the JitterAddingBackoff parameters from absolute values to rates

### LINKED ISSUE #619: Use percentage instead of absolute value as JitterAddingBackoff parameters
Small jitter values don't make much difference when using exponential backoff and the delay grows large. What do you think about using percentage instead of absolute value? /cc @imasahiro

### PR DESCRIPTION
**Pull request by minwoox**
Movtivation:
Small jitter values don't make much difference when the backoff delay grows large. We needed to change the parameters from absolute values to rates so it will work properly even when the delay is large.

Modifications:
- Change the parameters from absoulte values to rates

Result:
- JitterAddingBackoff works properly even when the delay is large
- Close #619

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/client/retry/Backoff.java [hunk 1]
```diff
@@ -84,19 +85,60 @@ static Backoff exponential(long initialDelayMillis, long maxDelayMillis, double
     }
 
     /**
-     * Returns a {@link Backoff} that provides a delay that increases using
+     * Returns a {@link Backoff} that provides a jitter value for the backoff delay using
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/client/retry/Backoff.java]: How about:

- provides a jitter value for the backoff delay -> adds a random jitter value to the original delay

File: core/src/main/java/com/linecorp/armeria/client/retry/Backoff.java [hunk 2]
```diff
@@ -84,19 +85,60 @@ static Backoff exponential(long initialDelayMillis, long maxDelayMillis, double
...
      * <a href="https://www.awsarchitectureblog.com/2015/03/backoff.html">full jitter</a> strategy.
+     * The jitter value will be multiplied to backoff delay to calculate the bound. For example, if the
+     * calculated delay using {@link Backoff#exponential(long, long)} is 1000 millis and the provided
+     * jitter value is 0.3, the ultimate backoff delay will be chosen between 1000 * (1 - 0.3)
+     * and 1000 * (1 + 0.3) by {@link ThreadLocalRandom}.
+     * The rate value should be between {@code 0.0f} and {@code 1.0f}.
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/client/retry/Backoff.java]: - The jitter value will be multiplied to backoff delay to calculate the bound. -> Could you rephrase this so it's a little bit easier to understand?
- calculated delay using ... is 1000 millis -> the delay returned by `{@link #nextDelayMillis(...)}` is 1000 milliseconds
- 0.0f, 1.0f -> 0.0 and 1.0 (f is for float and we use double here)

File: core/src/main/java/com/linecorp/armeria/client/retry/Backoff.java [hunk 3]
```diff
@@ -84,19 +85,60 @@ static Backoff exponential(long initialDelayMillis, long maxDelayMillis, double
...
     }
 
     /**
-     * Returns a {@link Backoff} that provides a delay that increases using
+     * Returns a {@link Backoff} that provides a jitter value for the backoff delay using
      * <a href="https://www.awsarchitectureblog.com/2015/03/backoff.html">full jitter</a> strategy.
+     * The {@code minJitterRate} and {@code maxJitterRate} will be multiplied to backoff delay to
+     * calculate the bound. For example, if the calculated delay using {@link Backoff#exponential(long, long)}
+     * is 1000 millis and the {@code minJitterRate} is -0.2, {@code maxJitterRate} is 0.3, the ultimate
+     * backoff delay will be chosen between 1000 * (1 - 0.2) and 1000 * (1 + 0.3) by {@code randomSupplier}.
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/client/retry/Backoff.java]: `by {@code randomSupplier}` -> Probably OK to remove this given that we have `@param` tag that explains it?

**Comment by trustin** [comment]
Thanks, @minwoox !

File: core/src/test/java/com/linecorp/armeria/client/retry/BackoffTest.java [hunk 4]
```diff
@@ -62,11 +60,22 @@ public void exponential() throws Exception {
...
-        Random random = new Random(1);
-        Backoff backoff = Backoff.fixed(100).withJitter(-50, 50, () -> random);
-        assertThat(backoff.nextDelayMillis(1)).isEqualTo(120);
-        assertThat(backoff.nextDelayMillis(2)).isEqualTo(139);
-        assertThat(backoff.nextDelayMillis(3)).isEqualTo(94);
+        int maxAttempts = 200;
+        Backoff backoff = Backoff.fixed(1000).withJitter(0.3).withMaxAttempts(maxAttempts);
+
+        for (int i = 1; i < maxAttempts; i++) {
+            assertThat(backoff.nextDelayMillis(i)).isBetween(700L, 1300L);
```
Comment (anuraaga) [inline core/src/test/java/com/linecorp/armeria/client/retry/BackoffTest.java]: Why change the style of the test from a seeded Random and fixed assertions? This doesn't seem to be a good test anymore, e.g., it's possible for a bug to cause no jitter to ever be added, so the value returned is always 1000 which still passes this test despite being broken.

Comment (minwoox) [inline core/src/test/java/com/linecorp/armeria/client/retry/BackoffTest.java]: Thank you for that point that I missed. I will make another PR for that.
